### PR TITLE
Exporter Landing page

### PR DESCRIPTION
**Pull request by SuperQ**
Add a basic exporter landing page.

### CODE REVIEW

File: web/web.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	config LandingConfig
+}
+
+func NewHandler(c LandingConfig) *handler {
+	return &handler{
+		config: c,
+	}
+}
+
+func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```
Comment (roidelapluie) [inline web/web.go:41]: You should serve the files here

Comment (SuperQ) [reply web/web.go:41]: Yup, I know, I'm going to work on finishing this soon. I'm probably going to more directly integrate this into the https handler code.

**Comment by LeviHarrison** [comment]
Might be nice to be able to "highlight" a few important metrics so that the user doesn't have to go digging in the `/metrics` page (eg. the Memcached exporter has the memory usage displayed).

**Comment by SuperQ** [comment]
It's still really ugly, but it works.

image

**Comment by SuperQ** [comment]
After this, I can add some more of the boiler plate stuff automatic in the web and kingpinflag packages.

File: web/generated_assets.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !genassets
+// +build !genassets
+
+package web
```
Comment (Nexucis) [inline web/generated_assets.go:17]: maybe it would be worth to add a comment explaining this file is generated and shouldn't be modified.

Comment (Nexucis) [reply web/generated_assets.go:17]: like 
`// THIS FILE HAS BEEN GENERATED, DO NOT EDIT.`

**Comment by Nexucis** [comment]
> @LeviHarrison That would be interesting, we could possibly pull a `prometheus.Describe()` to get the metric TYPE/HELP list.

if at some point we would like something a bit less home made, we should create a simple React app. That might help also to have something less ugly ;)

File: web/generated_assets.go (hunk 2 above)
Comment (SuperQ) [reply web/generated_assets.go:17]: Yup, that's a good idea.

**Comment by SuperQ** [comment]
Yup, I would love to have something less hand-rolled. But I don't want to get _too_ fancy. This is just a simple exporter landing page. I would like to keep it as a single HTML page if possible.

**Comment by Nexucis** [comment]
yeah that's the risk to go with React, it won't be that simple anymore and as a side effect potentially the binary size increase more than what we would like for a simple page.

**Comment by Nexucis** [comment]
so It's a trade off.
It's just if at some point we want that involved javascript for example we should go with React. Until that, we should be good with what you are doing @SuperQ

**Review by Nexucis — APPROVED** [review]

**Comment by SuperQ** [comment]
Yea, I want to avoid JS if possible. It _might_ be useful for some exporters that have forms for validation. Like blackbox_exporter. But exporter endpoints are meant to be minimal by design. I don't want to hear about any security issues for this feature.

**Review by roidelapluie — COMMENTED** [review]
I will review this shortly. We should avoid the generate assets and use embed.

**Comment by SuperQ** [comment]
Oh, I wasn't aware of `embed`. I'll update to that, looks way easier.

**Comment by SuperQ** [comment]
Oh, I wasn't aware of `embed`. I'll update to that, looks way easier.

**Comment by SuperQ** [comment]
Ok, I made some improvements to the look.

image

**Comment by Nexucis** [comment]
> We should avoid the generate assets and use embed.

Ah yeah I forgot about that everytime.

**Comment by Nexucis** [comment]
looks definitely better with `embed` !!

**Comment by dswarbrick** [comment]
> Ok, I made some improvements to the look.

Can we maybe embed some minimal CSS to use a sans-serif font, so that it looks a bit more modern?

**Comment by dswarbrick** [comment]
> It's just if at some point we want that involved javascript for example we should go with React.

Please keep in mind the difficulty of packaging some transpiled web scripts, especially in Debian, which has quite strict policies regarding that. For example, the Alertmanager web UI is already completely stripped in the Debian prometheus-alertmanager package, due to its gargantuan list of dependencies that would need to be packaged also.

I see no real justification to bloat a simple splash page with scripts.

**Comment by SuperQ** [comment]
@dswarbrick How about this:

image

**Pull request merged**